### PR TITLE
Fix gpu_id

### PR DESCRIPTION
**Pull request by fatwir**
This Pull Request enhances the GPU selection mechanism based on memory utilization. Specifically, it implements the following changes:

1. **GPU ID Selection**: 
   - The GPU ID will be selected based on memory utilization. If the memory utilization of a given GPU is less than 20%, the first MIG ID under that GPU will be selected.
   
2. **Limitations**:
   - While `nvidia-smi` does support a MIG option, I currently do not have the necessary permissions to access this feature.
   - Detailed information about the MIG-partitioned GPU utilization can be retrieved using [`dcgmi`](https://developer.nvidia.com/dcgm). However, this tool is not installed and cannot be utilized.

3. **Current Approach**:
   - For the time being, a manual search through the output of `nvidia-smi -L` is performed to select the appropriate GPU IDs.

### CODE REVIEW

File: src/br/analysis/prereq.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+)
+from br.models.utils import get_all_configs_per_dataset
+
+gc.collect()
+torch.cuda.empty_cache()
+
+# Based on the utilization, set the GPU ID
+
+
+def get_gpu_info():
```
Comment (vianamp) [inline src/br/analysis/prereq.py:25]: Wonder if we should use something like this instead: https://github.com/anderskm/gputil

**Review by vianamp — APPROVED** [review]
I did not test it.

**Comment by ritvikvasan** [comment]
this PR seems to be doing a lot more than this? did you mean to merge into the other nb-> python file branch?

**Comment by fatwir** [comment]
I had forked from [this](https://github.com/AllenCell/benchmarking_representations/tree/scripts_features_rebased) branch and updated the _setup_gpu function in the scripts!

**Comment by ritvikvasan** [comment]
can you merge into that branch then so its easier to know what you added?

File: src/br/analysis/prereq.py (hunk 1 above)
Comment (fatwir) [reply src/br/analysis/prereq.py:25]: Yeah this looks good! This will need to be updated for the MIG partitioned GPUs though!

**Comment by ritvikvasan** [comment]
I should have clarified here I didn't mean to drop this PR and just merge your changes, but to just base your PR off that branch

**Review by ritvikvasan — APPROVED** [review]
Much better, thanks!

**Pull request merged**